spec/services/course/user_invitation_service_spec.rb

```diff
@@ -0,0 +1,228 @@
+require 'rails_helper'
+
+RSpec.describe Course::UserInvitationService do
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    def temp_csv_from_attributes(records)
+      Tempfile.new(File.basename(__FILE__, '.*')).tap do |file|
+        file.write(CSV.generate do |csv|
+          csv << [:name, :email]
+          records.each do |user|
+            csv << [user.name, user.email]
+          end
+        end)
+        file.rewind
+      end
+    end
+
+    let(:course) { build(:course) }
+    subject { Course::UserInvitationService.new(course) }
+
+    let(:existing_users) do
+      (1..2).map do
+        create(:instance_user).user
+      end
+    end
+    let(:existing_user_attributes) do
+      existing_users.map do |user|
+        { name: user.name, email: user.email }
+      end
+    end
+    let(:new_users) do
+      (1..2).map do
+        create(:user)
+      end
+    end
+    let(:new_user_attributes) do
+      new_users.map do |user|
+        { name: user.name, email: user.email }
+      end
+    end
+    let(:users) { existing_users + new_users }
+    let(:user_attributes) { existing_user_attributes + new_user_attributes }
+
+    describe '#invite' do
+      def verify_users
```

Comment:
Assignment Branch Condition size for verify_users is too high. [43.09/15]<br>Method has too many lines. [16/10]